File: 031-连续子数组的最大和/031-FindGreatestSumOfSubArray.cpp
class Solution {
public:
    int FindGreatestSumOfSubArray(vector<int> array) {
        vector<int> dp(array.size());
        dp[0]=array[0];
        int ans=dp[0];
        for(int i =1;i<array.size();i++){
            dp[i]=dp[i-1]+array[i]>array[i]?dp[i-1]+array[i]:array[i];
            ans=dp[i]>ans?dp[i]:ans;
        }
        return ans;
    }
};

File: 031-连续子数组的最大和/031-FindGreatestSumOfSubArray.py
# -*- coding:utf-8 -*-
class Solution:
    def FindGreatestSumOfSubArray(self, array):
        # write code here
        dp=[0]*len(array)
        dp[0]=array[0]
        ans=float('-inf')
        for i in range(1,len(array)):
            dp[i]=max(dp[i-1]+array[i],array[i])
            ans=max(ans,dp[i])
        return ans
